### PR TITLE
chore(release): replace SHA-512 with SHA-256 for release checksums

### PR DESCRIPTION
**Pull request by starwarsfan**
## Summary

- Switch from SHA-512 to SHA-256 for all release artifact checksums — matches the hash GitHub natively displays beside each release asset, eliminating user confusion about which checksum to use
- SHA files are no longer uploaded as separate release assets; checksums are embedded in the release notes body under a flat sorted `### Checksums (SHA-256)` section (app bundle → amd64 → arm64)
- `obs-update` no longer downloads a separate `.sha256` file; it parses the SHA-256 directly from the release notes body and verifies the downloaded bundle with `sha256sum -c -`; older releases without a checksum in their notes get a warning and continue
- Updated LXC download instructions in both `README.md` and `README.de.md` to reflect the new SHA-256 workflow and new screenshots
- Updated existing unit test to assert the new verification pattern

## Test plan

- [ ] `tests/unit/test_lxc_template_updater.py` passes (verified locally)
- [ ] No `sha512` references remain in the workflow
- [ ] Release notes checksum section renders in correct order: app bundle, amd64, arm64
- [ ] `obs-update` on an existing LXC can update to a new release and the SHA-256 verification passes
- [ ] `obs-update` against an older release (no checksum in notes) warns and continues

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .github/workflows/lxc-template.yml [hunk 1]
```diff
@@ -185,38 +185,41 @@ jobs:
...
           curl -fL "$BUNDLE_URL" -o "$TMP/$BUNDLE_FILENAME"
-          echo "Downloading $CHECKSUM_URL ..."
-          curl -fL "$CHECKSUM_URL" -o "$TMP/$CHECKSUM_FILENAME"
 
-          (cd "$TMP" && sha512sum -c "$CHECKSUM_FILENAME")
+          if [[ -n "$EXPECTED_SHA256" ]]; then
+              echo "Verifying integrity ..."
+              echo "${EXPECTED_SHA256}  ${BUNDLE_FILENAME}" | (cd "$TMP" && sha256sum -c -)
+          else
+              echo "Warning: no SHA-256 found in release notes, skipping integrity check."
```
Comment (Micsi) [inline .github/workflows/lxc-template.yml]: Das sollte aus Security-Sicht fail-closed sein. `obs-update` laeuft als Root, laedt ein App-Bundle herunter und extrahiert danach Code/Assets in die Installation. Wenn die SHA-256 nicht aus den Release Notes geparst werden kann, sollte der Update abbrechen statt ohne Integritaetspruefung fortzufahren. Fuer alte Releases waere ein expliziter Legacy-Pfad besser als ein generelles `skip integrity check`.

File: .github/workflows/lxc-template.yml [hunk 2]
```diff
@@ -597,36 +600,22 @@ jobs:
...
-              + lxc_section
-              + bundle_section
-              + marker
-          )
+          section = "### Checksums (SHA-256)\n\n"
+          for name, cs in entries:
+              section += f"*{name}*:\n```\n{cs}\n```\n\n"
+          section += marker
+
           print(body.replace(marker, section), end='')
```
Comment (Micsi) [inline .github/workflows/lxc-template.yml:654]: Der Replace laesst den Marker im Body stehen und entfernt vorhandene Checksum-Sektionen nicht. Bei einem erneuten Release-Job kann dadurch eine zweite Checksum-Sektion entstehen; `obs-update` matched dann per Regex wahrscheinlich die erste `app-bundle`-Checksumme und koennte gegen einen stale Hash pruefen. Bitte idempotent machen, z. B. vorhandenen generierten Block ersetzen oder vor dem Edit sicherstellen, dass der Marker genau einmal und ohne alten Block vorhanden ist.

File: README.md [hunk 3]
```diff
@@ -69,15 +69,26 @@ The LXC template contains a complete Ubuntu 26.04 system with **open bridge serv
...
+6. Select **SHA256** as the hash algorithm.
+7. On the [releases page](../../releases/latest), copy the checksum of the desired template from the **Checksums** section using the copy button:
+  
+   ![ProxmoxDownloadFromURL](docs/Release-Asset-Checksums.png)
+
+   Note: If you copy the checksum directly from the column next to the asset, remove the `SHA256:` prefix, as Proxmox does not expect it!
+8. Back in the Proxmox web interface, paste the copied hash into the **Checksum** field.
+9. It should now look like this, for example:
+
+   ![ProxmoxDownloadFromURL](docs/ProxmoxDownloadFromURL.png)
```
Comment (Micsi) [inline README.md:89]: Korrektur/Einordnung: Der funktionale `[releases page](../../releases/latest)`-Link wird im Upstream-README korrekt auf `abeggled/openbridgeserver/releases/latest` gerendert. Das hier ist daher kein falscher Download-Link. Der Punkt betrifft nur den statischen Screenshot: Im Proxmox-URL-Feld ist sichtbar ein Contributor-Fork (`github.com/starwarsfan/openbridgeserver/...`) zu sehen. Das wuerde ich als Doku-/Polish-Fix ersetzen oder neutralisieren, aber nicht als Security-Blocker werten.

File: README.md [hunk 4]
```diff
@@ -69,15 +69,26 @@ The LXC template contains a complete Ubuntu 26.04 system with **open bridge serv
...
 2. In the Proxmox web interface, navigate to **Datacenter → Storage → local → CT Templates**.
 3. Click **Download from URL**.
 4. Paste the copied URL and click **Query URL**.
-5. Select **SHA512** as the hash algorithm.
-6. Paste the copied hash.
-7. Click **Download**.
+5. If not already enabled, activate **Advanced** in the bottom right of the popup.
+6. Select **SHA256** as the hash algorithm.
+7. On the [releases page](../../releases/latest), copy the checksum of the desired template from the **Checksums** section using the copy button:
+
```
Comment (Micsi) [inline README.md]: Hier ist trailing whitespace; `git diff --check` schlaegt dadurch fehl. Bitte entfernen.

File: README.de.md [hunk 5]
```diff
@@ -69,15 +69,27 @@ Das LXC-Template enthält ein vollständiges Ubuntu 26.04-System mit **open brid
...
 4. Die kopierte URL einfügen und auf **Query URL** klicken.
-5. Als Hash-Algorithmus **SHA512** auswählen.
-6. Den kopierten Hash einfügen.
-7. Auf **Download** klicken.
+5. Wenn nicht bereits aktiviert, im Popup unten rechts **Advanced** aktivieren.
+6. Als Hash-Algorithmus **SHA256** auswählen.
+7. Auf der [Release-Seite](../../releases/latest) im Abschnitt **Checksums** via Copy-Button die Checksumme des gewünschten Templates kopieren:
+
+   ![ProxmoxDownloadFromURL](docs/Release-Asset-Checksums.png)
+
```
Comment (Micsi) [inline README.de.md]: Hier ist ebenfalls trailing whitespace; bitte entfernen, damit `git diff --check` sauber bleibt.

**Review by Micsi — CHANGES_REQUESTED** [review]
Danke fuer den PR. Die Richtung, GitHub-nativ sichtbare SHA-256-Checksummen fuer die LXC-Releases zu verwenden, passt grundsaetzlich zum Scope. In der aktuellen Form wuerde ich ihn aber noch nicht mergen, weil der Update-Pfad bei fehlender/ungeparster Checksumme fail-open wird und der Release-Body als neue Checksum-Quelle robuster behandelt werden sollte.

Zusaetzlich ist der Scope im PR-Text groesser als die Umsetzung: Nightly-LXC veroeffentlicht weiterhin .sha512-Assets und tools/_lxc-inner.sh erzeugt/verwendet weiterhin SHA-512 fuer lokale Bundle-/LXC-Builds. Entweder diese Pfade mit umstellen oder Titel/Beschreibung auf versionierte LXC-Tag-Releases eingrenzen.

Geprueft: PR #832, Head 2c3054c0a8a248ceb2ce8108983fd208af0d4946, Worktree sauber; tools/with-venv pytest tests/unit/test_lxc_template_updater.py ist gruen. git diff --check meldet noch trailing whitespace in README.md und README.de.md.

**Comment by Micsi** [comment]
@starwarsfan Was hältst du davon, actionLint für Änderungen an GitHub Workflows ins Ci aufzunehmen?

**Comment by starwarsfan** [comment]
> @starwarsfan Was hältst du davon, actionLint für Änderungen an GitHub Workflows ins Ci aufzunehmen?

Gute Idee, sehr gerne!

File: README.md (hunk 3 above)
Comment (starwarsfan) [reply README.md:89]: Worauf beziehst Du Dich hier? Wo ist hier ein allfälliger Security-Blocker?

File: README.md (hunk 4 above)
Comment (starwarsfan) [reply README.md]: Wo ist das Problem hier? Trailing whitespace ist unschön, klar aber wer schlägt hier wie und wo genau fehl?

**Comment by abeggled** [comment]
@codex review

**Comment by abeggled** [comment]
@codex review

**Comment by abeggled** [comment]
@codex review

File: .github/workflows/nightly-lxc.yml [hunk 6]
```diff
@@ -110,7 +110,7 @@ jobs:
           fi
 
           mapfile -t ASSETS < <(find artifacts -type f \
-            \( -name '*.tar.zst' -o -name '*.tar.gz' -o -name '*.sha512' \) | sort)
+            \( -name '*.tar.zst' -o -name '*.tar.gz' -o -name '*.sha256' \) | sort)
```
[the remark this answers is not in the document]
Comment (starwarsfan) [reply .github/workflows/nightly-lxc.yml:113]: The P1 Codex concern is actually already solved in this PR — look at the workflow diff. Both  
  checksums are published for every app bundle:                                                 
                                                                                                
  path: |                                                                                       
    app_bundle_file                                                                             
    app_bundle_file.sha256                                                                      
    app_bundle_file.sha512   ← still there                                                      
                                                                                                
  And _lxc-inner.sh likewise generates both. So old obs-update scripts that look for .sha512 can
  still update to the first new release. The P1 finding is invalid for this PR as written.      
                                                                                                
  Your "bridge version" idea is also not needed — the PR already implements a cleaner approach: 
  - New releases publish both sha256 and sha512 (backward compat for old updaters)              
  - The new obs-update embedded in those releases prefers sha256 from the release notes body,   
  falls back to the .sha512 asset for older rollback targets                                    
  - So both directions — forward and rollback — work across the migration without any version   
  cap logic                                                                                     
                                                                                                
  What IS a valid remaining concern is the P2 nightly issue: nightly-lxc.yml now uploads only   
  *.sha256 artifacts (not *.sha512), but nightly release bodies don't get the ### Checksums     
  (SHA-256) injection that the release workflow adds. So a new obs-update on a nightly would    
  find neither an embedded sha256 nor a sha512 asset and abort. That's a narrower edge case, but
  real if anyone tests against nightlies.                                                       
                                                                                                
  The other Codex items (P3 clean step, local builder fallback) are already fixed in the diff.  
                                                                                                
  Bottom line: The PR is solid. The P1 is a false alarm. The user's bridge-version approach     
  would have worked but adds unnecessary complexity — the dual-publish strategy here is simpler 
  and more transparent. The one thing worth a look is whether nightlies should also get the     
  sha256 checksum block injected into their release body, or keep publishing sha512 as well.

**Review by abeggled — APPROVED** [review]
Merci Yves
